launcher/launcher_test.go

```diff
@@ -716,6 +716,33 @@ func TestThatEndpointsFallBackCorrectly(t *testing.T) {
 			metricsEndpoint: "localhost:4317",
 			metricsInsecure: false,
 		},
+		{
+			name: "set grpc endpoint with https scheme and no port, add port as helper",
+			config: newConfig(
+				WithExporterEndpoint("https://generic-url"),
+				WithExporterProtocol("grpc"),
+			),
+			tracesEndpoint:  "generic-url:443",
+			metricsEndpoint: "generic-url:443",
+		},
+		{
+			name: "set grpc endpoint with https scheme and port, no update to port",
+			config: newConfig(
+				WithExporterEndpoint("https://generic-url:1234"),
+				WithExporterProtocol("grpc"),
+			),
+			tracesEndpoint:  "generic-url:1234",
```

Comment:
Using `:1234` in this test now to confirm saving of provided port and not a fallback to default `:4317`